Ginger/GingerCoreNET/ValueExpressionLib/ValueExpression.cs

```diff
@@ -248,8 +263,59 @@ private void EvaluateFlowDetails()
                     Reporter.ToLog(eLogLevel.ERROR, string.Format("Failed to evaluate flow details expression '{0}'", match.Value), ex);
                 }
             }
+
+            foreach (Match match in matchesExecutionJsonData)
+            {
+                try
+                {
+                    ReplaceExecutionJsonDataDetails(match.Value);
+                }
+                catch (Exception ex)
+                {
+                    mValueCalculated = mValueCalculated.Replace(match.Value, string.Format("['{0}' Expression is not valid, Error:'{1}']", match.Value, ex.Message));
+                    Reporter.ToLog(eLogLevel.ERROR, string.Format("Failed to evaluate flow details expression '{0}'", match.Value), ex);
+                }
+            }
         }
+        
+        private void ReplaceExecutionJsonDataDetails(string josnExpression)
+        {
+            //Select fields from selected template configuration
+            ObservableList<HTMLReportConfiguration> HTMLReportConfigurations = WorkSpace.Instance.SolutionRepository.GetAllRepositoryItems<HTMLReportConfiguration>();
+            HTMLReportConfiguration defaultTemplate = null;
+            if (mContext.RunsetAction != null)
+            {
+                RunSetActionSendDataToExternalSource runSetAction = (RunSetActionSendDataToExternalSource)mContext.RunsetAction;
+                defaultTemplate = HTMLReportConfigurations.Where(x => (x.ID == runSetAction.selectedHTMLReportTemplateID)).FirstOrDefault();
+                if (defaultTemplate == null && runSetAction.selectedHTMLReportTemplateID == 100)
+                {
+                    defaultTemplate = HTMLReportConfigurations.Where(x => (x.IsDefault == true)).FirstOrDefault();
+                }
+            }
+            else 
+            {
+                defaultTemplate = HTMLReportConfigurations.Where(x => (x.IsDefault == true)).FirstOrDefault();
+            }
 
+            //Get Last Execution details
+            LiteDbRunSet liteDbRunSet = null;
+            var loggerMode = WorkSpace.Instance.Solution.LoggerConfigurations.SelectedDataRepositoryMethod;
+            if (loggerMode == ExecutionLoggerConfiguration.DataRepositoryMethod.LiteDB)
+            {
+                LiteDbManager dbManager = new LiteDbManager(new ExecutionLoggerHelper().GetLoggerDirectory(WorkSpace.Instance.Solution.LoggerConfigurations.CalculatedLoggerFolder));
+                liteDbRunSet = dbManager.GetLatestExecutionRunsetData("");
+            }
+            else
+            {
+                mValueCalculated = mValueCalculated.Replace(josnExpression, string.Format("['{0}' Expression Supported only in case the selected Execution Logger type is DB.]", josnExpression));
+                return;
+            }
+
+            LiteDBRepository liteDBRepository = new LiteDBRepository();
```

Comment:
Why to create new liteDBRepository object. you can retrieve existing object from workspace